test/snippet/core/detail/all_of_in_pack.cpp

```diff
@@ -0,0 +1,37 @@
+#include <type_traits>
+
+#include <seqan3/core/detail/pack_algorithm.hpp>
+#include <seqan3/core/type_list/type_list.hpp>
+#include <seqan3/core/debug_stream.hpp>
+
+int main()
+{
+    // With c++20 you could also write it like this
+    // auto fn = []<typename type>(std::type_identity<type>)
+    // {
+    // ...
+    // };
+    auto fn = [](auto id)
+    {
+        // id is the original type not wrapped in std::type_identity.
+        using id_t = decltype(id);
+
+        if constexpr(std::is_same_v<id_t, bool>)
+            return true;
+        else if constexpr(std::is_same_v<id_t, int>)
+            return true;
+        else if constexpr(std::is_same_v<id_t, double>)
+            return true;
+        else
+            return false;
+    };
```

Comment:
I think this snippet could also have a snippet where you ask for certain values.

```c++
        if constexpr(std::is_same_v<id_t, bool>)
            return id == false;
        else if constexpr(std::is_same_v<id_t, int>)
            return id == 3;
        else if constexpr(std::is_same_v<id_t, double>)
            return std::abs(id - 1.2) < 0.00001;
        else
            return false;
```